### PR TITLE
Tenant provision

### PR DESCRIPTION
**Pull request by elv-wayne**
Eg:
Task 3 of ticket #53
```
./elv-live tenant_provision test6 --verbose
Tenant Provision
tenantName: test6
verbose: true
Network: demov3
kmsId: ikms27hsDr6QtmWnJKVFkNzroSGEgjjq

Tenant Admin Groups ID:

         iten2tW1wPCwmdHmXJDPx88h34mhKGAe

Access Groups:

        Organization Admins Group: 0x87838c5638d56589bebf5aeccf9702bb5d6219db
        Content Admins Group: 0xa6a2bef249a16cf58957629f02f3638c846bbdb4
        Content Users Group: 0x2d504452cbb9af18f0426c7a25b79ec0e40ea283

Tenant Types:

        test6 - Title: iq__SA3isZngvVc5buQzVFggj7MgVoG
        test6 - Title Collection: iq__23SmkPxn7DfbzUDJr5qQoqa4eRCR
        test6 - Title Master: iq__2V4PMNo25sUyDwwBWxDvNd7B8gt6
        test6 - Permissions: iq__2HhYEa7NtArjZa8kTrLncZKPdHba
        test6 - Channel: iq__44qzB8c4iDtedUtqsjLJSp89tnps
        test6 - Live Stream: iq__3ypY62bPFQYWnpk4ESTvk7q4e9MW
        test6 - Eluvio LIVE Drop Event Site: iq__48w4fWb2CGnzkiCTFRScSfcKY5TX
        test6 - Eluvio LIVE Marketplace: iq__V4DekRHeFtFSaarhXEM1mwpg3DS
        test6 - Eluvio LIVE Tenant: iq__3DewzqFi9hAqcctCTWSXZeoLC3VZ
        test6 - NFT Collection: iq__cj337B5UiXzdYDEWRV8DXHineGS
        test6 - NFT Template: iq__3bzKstB8ixF2gPBtxJfqB6BWVEFr

Tenant Libraries:

        test6 - Properties: ilibc6LcQ4C3ohTvUm8w4xBXYVBeVu8
        test6 - Title Masters: ilibZYxGHqyX5okGktmXv5TKqfgtaTV
        test6 - Title Mezzanines: ilib4XDhpiVe6rsfM9rmiTncBHsX2v1H
        test6 - NFT Templates: ilib28ZryMMJjp41VAErV5qDZP8m26tG

Site Object: 

        Site - test6: iq__3yvYdaW1sVs1d95rqRAXM5s5CUz9


tenantTypes:
  titleTypeId: iq__SA3isZngvVc5buQzVFggj7MgVoG
  titleCollectionTypeId: iq__23SmkPxn7DfbzUDJr5qQoqa4eRCR
  masterTypeId: iq__2V4PMNo25sUyDwwBWxDvNd7B8gt6
  permissionsTypeId: iq__2HhYEa7NtArjZa8kTrLncZKPdHba
  channelTypeId: iq__44qzB8c4iDtedUtqsjLJSp89tnps
  streamTypeId: iq__3ypY62bPFQYWnpk4ESTvk7q4e9MW
libraries:
  propertiesLibraryId: ilibc6LcQ4C3ohTvUm8w4xBXYVBeVu8
  mastersLibraryId: ilibZYxGHqyX5okGktmXv5TKqfgtaTV
  mezzanineLibraryId: ilib4XDhpiVe6rsfM9rmiTncBHsX2v1H
  nftLibraryId: ilib28ZryMMJjp41VAErV5qDZP8m26tG
groups:
  tenantAdminGroupAddress: '0x87838c5638d56589bebf5aeccf9702bb5d6219db'
  contentAdminGroupAddress: '0xa6a2bef249a16cf58957629f02f3638c846bbdb4'
  contentUserGroupAddress: '0x2d504452cbb9af18f0426c7a25b79ec0e40ea283'
siteId: iq__3yvYdaW1sVs1d95rqRAXM5s5CUz9

```

### CODE REVIEW

File: src/Provision.js [hunk 1]
```diff
@@ -0,0 +1,313 @@
+/* eslint-disable no-console */
+
+const liveTypes = [
```
Comment (elv-kevin) [inline src/Provision.js:3]: we can actually get rid of these specs for live types, those are contained in asset manager now.

File: src/Provision.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+  let groups = {
+    tenantAdminGroupAddress,
+    contentAdminGroupAddress,
+    contentUserGroupAddress
+  };
+
+  /* Content Types - Create Title, Title Collection and Production Master and add each to the groups */
+
+  const typeMetadata = {
```
Comment (elv-kevin) [inline src/Provision.js:109]: can you make it so Production Master doesn't have display app set? Michael filed an issue for this in elv-client-js

https://github.com/eluv-io/elv-client-js/issues/155

**Review by elv-kevin — APPROVED** [review]

File: src/Provision.js (hunk 1 above)
Comment (elv-wayne) [reply src/Provision.js:3]: So get rid of this code as well?
```
  for (let i = 0; i < liveTypes.length; i++) {
    const typeId = await client.CreateContentType({
      name: `${tenantName} - ${liveTypes[i].name}`,
      metadata: {
        ...typeMetadata,
        public: {
          ...typeMetadata.public,
          title_configuration: liveTypes[i].spec
        }
      }
    });

    await SetObjectPermissions(client, typeId, tenantAdminGroupAddress, contentAdminGroupAddress, contentUserGroupAddress);

    if (debug) {
      console.log(`\t${tenantName} - ${liveTypes[i].name}: ${typeId}`);
    }
  }
  ```

File: src/Provision.js (hunk 2 above)
Comment (elv-wayne) [reply src/Provision.js:109]: Remove it from the typeMetadata like this?
```
  const typeMetadata = {
    bitcode_flags: "abrmaster",
    bitcode_format: "builtin",
    public: {
      "eluv.manageApp": "default",
    }
  };
 ```

File: src/Provision.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+  await Promise.all(promises);
+};
+
+const SetObjectPermissions = async (client, objectId, tenantAdmins, contentAdmins, contentUsers) => {
+  let promises = [
+    // Tenant admins
+    client.AddContentObjectGroupPermission({objectId, groupAddress: tenantAdmins, permission: "see"}),
+    client.AddContentObjectGroupPermission({objectId, groupAddress: tenantAdmins, permission: "access"}),
+    client.AddContentObjectGroupPermission({objectId, groupAddress: tenantAdmins, permission: "manage"}),
```
Comment (elv-serban) [inline src/Provision.js:48]: We only need manage!  This is an ordered permission level so manage > acccess > see

File: src/Provision.js (hunk 1 above)
Comment (elv-wayne) [reply src/Provision.js:6]: We decided just to leave it there

**Pull request merged**